app/controllers/api/v1/rooms_controller.rb

```diff
@@ -16,6 +16,8 @@ class RoomsController < ApiController
         ensure_authorized('ManageRooms', friendly_id: params[:friendly_id])
       end
 
+      after_action :access_code_checker, only: :create
```

Comment:
I'm not sure I like the `after_action` / `after_create`. I'd prefer if we move this logic directly into the `create` action instead

Reply:
Since you're depending on the action initializing a `@new_room` for you

Reply:
Yeah, well this would solve my current issue with Rspec. I'm having a hard time accessing the after_action in specs.